packages/cursorless-engine/src/generateSpokenForm/generateSpokenForm.test.ts

```diff
@@ -55,6 +56,31 @@ suite("Generate spoken forms", () => {
     assert(spokenForm.type === "success");
     assert.equal(spokenForm.spokenForms, "bar");
   });
+
+  test("generate spoken form for command with placeholder", () => {
+    const generator = new SpokenFormGenerator(spokenFormMap);
+
+    const spokenForm = generator.processCommand({
+      action: {
+        name: "setSelection",
+        target: {
+          type: "primitive",
+          modifiers: [
+            { type: "containingScope", scopeType: { type: "paragraph" } },
+          ],
+          mark: {
+            type: "placeholder",
+            index: 0,
+          },
+        },
+      },
+      version: LATEST_VERSION,
+      usePrePhraseSnapshot: false,
+    });
+
+    assert(spokenForm.type === "success");
+    assert.equal(spokenForm.spokenForms, "take block <target>");
```

Comment:
Note the similarity to the cheatsheet here. I had a fantasy of using this machinery to auto generate parts of cheatsheet and docs

Reply:
Neat!